### PR TITLE
[CI] Update to Swift 6 CI

### PR DESCRIPTION
**Pull request by FranzBusch**
# Motivation

We just updated our CI matrix in NIO to only support 5.9, 5.10 and 6.

# Modification

This PR updates the trigger files in this repo. Since this repo was always 5.9+ this is easy.

# Result

Up to date CI

### CODE REVIEW

**Comment by czechboy0** [comment]
@FranzBusch 

```
[Invalid workflow file: .github/workflows/pull_request.yml#L10](https://github.com/apple/swift-openapi-runtime/actions/runs/11345355766/workflow)
error parsing called workflow
".github/workflows/pull_request.yml"
-> "apple/swift-nio/.github/workflows/soundness.yml@main"
: failed to fetch workflow: workflow was not found.
```

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -9,22 +9,15 @@ jobs:
...
         uses: apple/swift-nio/.github/workflows/soundness.yml@main
         with:
-            api_breakage_check_enabled: true
-            broken_symlink_check_enabled: true
-            docs_check_enabled: true
-            format_check_enabled: true
-            license_header_check_enabled: true
             license_header_check_project_name: "SwiftOpenAPIGenerator"
-            shell_check_enabled: true
-            unacceptable_language_check_enabled: true
```
Comment (simonjbeaumont) [inline .github/workflows/pull_request.yml:13]: Why are we removing these?

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -9,22 +9,15 @@ jobs:
         name: Soundness
         uses: apple/swift-nio/.github/workflows/soundness.yml@main
```
Comment (czechboy0) [inline .github/workflows/pull_request.yml:10]: Should this be switched to the reusable one?

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (czechboy0) [reply .github/workflows/pull_request.yml:13]: I assume because they're on by default?

File: .github/workflows/scheduled.yml [hunk 3]
```diff
@@ -9,9 +9,9 @@ jobs:
         name: Unit tests
         uses: apple/swift-nio/.github/workflows/unit_tests.yml@main
```
Comment (czechboy0) [inline .github/workflows/scheduled.yml:10]: Same here? https://github.com/apple/swift-openapi-runtime/pull/122/files#r1801118837

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (FranzBusch) [reply .github/workflows/pull_request.yml:13]: Correct. The default is `true` so no reason to pass it additionally

**Comment by FranzBusch** [comment]
@czechboy0 @simonjbeaumont the documentation pipeline seems to be failing since switching over to the reusable repo updated the toolchain for that workflow from 5.10 to 6.0. It looks like there are some new warnings now that need to be fixed. Could you help me with that?

File: .github/workflows/pull_request.yml [hunk 4]
```diff
@@ -7,24 +7,17 @@ on:
 jobs:
```
Comment (czechboy0) [inline .github/workflows/pull_request.yml:7]: Seems the indentation is off two lines higher, @FranzBusch

**Comment by czechboy0** [comment]
> @czechboy0 @simonjbeaumont the documentation pipeline seems to be failing since switching over to the reusable repo updated the toolchain for that workflow from 5.10 to 6.0. It looks like there are some new warnings now that need to be fixed. Could you help me with that?

Yes let me open a separate PR making those fixes first.

**Comment by czechboy0** [comment]
@FranzBusch https://github.com/apple/swift-openapi-runtime/pull/123

**Comment by czechboy0** [comment]
@FranzBusch Actually, we have a chicken-and-egg problem, you need to bring this tiny diff into your PR instead, I can't merge mine with green CI anyway: https://github.com/apple/swift-openapi-runtime/pull/123

**Comment by FranzBusch** [comment]
@czechboy0 Thanks for that. I slightly tweaked your commit to also avoid the `Return:` on the closure since that documents the wrong thing. I am wondering if that should actually be allowed.

**Comment by czechboy0** [comment]
@FranzBusch So the YAML linter still fails: `PR / Soundness / YAML lint check (pull_request)`

**Comment by FranzBusch** [comment]
@czechboy0 fixed!

**Comment by czechboy0** [comment]
> @czechboy0 fixed!

I still see it as failing 😬

**Comment by FranzBusch** [comment]
@czechboy0 Finally it's all green. Please do me the honours!

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (simonjbeaumont) [reply .github/workflows/pull_request.yml:13]: Heh, OK. We discussed why we wanted them in when we onboarded them in the earlier PR, but fine.

**Pull request merged**